### PR TITLE
rspec testing part is added

### PR DESCRIPTION
**Pull request by sumancrest0001**
Testing part with RSpec.
1. performed negative as well as positive tests for all the methods.
2. Updated the readme.md file

### CODE REVIEW

File: spec/main_spec.rb [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+    context 'when valid input is given' do
+      it 'result should be true' do
+        expect(test_arr.any? { |ele| ele < 20 }).to eql(true)
+      end
+    end
+
+    describe 'when input is empty' do
+      it 'result is false' do
+        expect([].any? { |ele| ele < 20 }).to eql(false)
```
Comment (maelfosso) [inline spec/main_spec.rb]: It's not `any?` but `*`my_any?*

File: spec/main_spec.rb [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+  describe '#multiply_els method' do
+    context 'my_inject method is implemented' do
+      it 'should give product of each element' do
+        expect(test_arr.multiply_els { |sum, ele| sum * ele }).to be 1620
+      end
+    end
+
+    context ' when one element is zero' do
+      let(:arr) { [1, 3, 0] }
```
Comment (maelfosso) [inline spec/main_spec.rb]: You could use the array define at the beginning here.
Why are you use another one ?

File: spec/main_spec.rb [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+      end
+    end
+
+    context 'when one element is zero' do
+      let(:arr) { [1, 3, 0] }
+      it 'should be equal to zero' do
+        expect(arr.multiply_els { |sum, ele| sum * ele }).to be 0
+      end
+    end
+  end
```
Comment (maelfosso) [inline spec/main_spec.rb]: You've already tested it below.
You could check, please.

**Review by maelfosso — CHANGES_REQUESTED** [review]
Hello coders,

Thanks for the job done :clap: 
Even if you show that you master writing test with RSpec, there are still a few thing to change.

- [ ] Please I have this error when running `rspec`
```ruby
$ rspec
.F....................

Failures:

  1) Enumerable #my_each_with_index when input is valid should display array of index
     Failure/Error: expect(newArr).to eql([0, 1, 2, 3, 4])
     
       expected: [0, 1, 2, 3, 4]
            got: [1, 3, 6, 9, 10]
     
       (compared using eql?)
     # ./spec/main_spec.rb:21:in `block (4 levels) in <top (required)>'

Finished in 0.04813 seconds (files took 0.16164 seconds to load)
22 examples, 1 failure

Failed examples:

rspec ./spec/main_spec.rb:19 # Enumerable #my_each_with_index when input is valid should display array of index
```

- [ ] Please compare the result with the result of the original operation

- [ ] Read the comments below to get the position of things to changes

- [ ] Fix Sticklers errors

Kindly make those changes and submit a new code review request.
Thanks.

**Comment by sumancrest0001** [comment]
Made all the required changes.

**Review by bolah2009 — CHANGES_REQUESTED** [review]
Can you please fix linter warnings/errors and submit another code review request?

The test looks good to me. :smile:

**Review by maelfosso — APPROVED** [review]
# Congratulations :clap: :smile: 

You can merge and fill the project completion form.
**The project is approved**
Good luck with the next project.

**Pull request merged**